lib/ddtrace/contrib/rails/framework.rb

```diff
@@ -26,7 +26,8 @@ module Framework
           tracer: Datadog.tracer,
           debug: false,
           trace_agent_hostname: Datadog::Writer::HOSTNAME,
-          trace_agent_port: Datadog::Writer::PORT
+          trace_agent_port: Datadog::Writer::PORT,
+          env: ::Rails.env
```

Comment:
In my particular case, yes, I'm using a single Datadog agent per host, with each host receiving and forwarding data from multiple applications (specifically via Docker/Kubernetes). The Rails convention is _not_ to have your `staging` and `production` environments use the same environment name. Instead you should have environments with different names that share configuration.

With that being said, maybe the easiest solution to handle your use-case is to make it possible to set `env:` to `nil` on the Rails side, forcing it to inherit `env` from your agent config.